unlock-app/src/components/content/event/LockDeploying.tsx

```diff
@@ -0,0 +1,101 @@
+import { Button } from '@unlock-protocol/ui'
+import Link from 'next/link'
+import { useRouter } from 'next/router'
+import { FiExternalLink as ExternalLinkIcon } from 'react-icons/fi'
+import { useUpdateMetadata } from '~/hooks/metadata'
+
+import {
+  AnimationContent,
+  DeployStatus,
+} from '~/components/interface/locks/Create/elements/CreateLockFormSummary'
+import { useConfig } from '~/utils/withConfig'
+import { TransactionDetails } from './NewEvent'
+import { useEffect } from 'react'
+import { MetadataFormData } from '~/components/interface/locks/metadata/utils'
+import { formDataToMetadata } from '~/components/interface/locks/metadata/utils'
+
+interface LockDeployingProps {
+  transactionDetails: TransactionDetails
+  lockAddress?: string
+  metadata?: MetadataFormData
+}
+
+export const LockDeploying = ({
+  transactionDetails,
+  lockAddress,
+  metadata,
+}: LockDeployingProps) => {
+  const config = useConfig()
+  const router = useRouter()
+  const { hash: transactionHash, network } = transactionDetails
+  const { mutateAsync: updateMetadata, isLoading: isMetadataUpating } =
+    useUpdateMetadata({
+      lockAddress,
+      network,
+    })
+
+  useEffect(() => {
+    const save = async (metadata: MetadataFormData) => {
+      await updateMetadata(formDataToMetadata(metadata!))
+    }
+    if (metadata) {
+      save(metadata)
+    }
+  }, [updateMetadata, metadata])
```

Comment:
move this to the previous component where user clicks the button? This can result in race condition at the moment.

Reply:
ok, but please explain about the race condition!

Reply:
Just mean there is no guarantee useEffect won't be called twice. It most likely won't but in rare case, can be since something can trigger a re-render from top. 

It is called twice in development and localhost for this reason so users are aware to not put network request without a cancellation/clean up function.

Reply:
fixed!